### PR TITLE
Extend `Command` and `Event` using mixins

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR replaces static utility methods from `Commands` and `Events` with default methods offered by `CommandMixin` and `EventMixin` interfaces.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, @dmdashenkov, PTAL.

File: server/src/test/java/io/spine/core/EventTest.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                .build();
+        tester.setDefault(StringValue.class, StringValue.getDefaultInstance())
+              .setDefault(EventContext.class, GivenEvent.context())
+              .setDefault(Version.class, Version.getDefaultInstance())
+              .setDefault(Event.class, Event.getDefaultInstance())
+              .setDefault(ThrowableMessage.class, defaultThrowableMessage);
+    }
+
+    @Nested
+    @DisplayName("obtain")
```
Comment (armiol) [inline server/src/test/java/io/spine/core/EventTest.java]: To conform to the English grammar rules, I would modify some of the display names.
Here is would say [Event should] `provide`.

File: server/src/test/java/io/spine/core/EventTest.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        private Event newEvent(CommandEnvelope command) {
+            StringValue producerId = StringValue.of(getClass().getSimpleName());
+            EventFactory ef = EventFactory.on(command, Identifier.pack(producerId));
+            return ef.createEvent(GivenEvent.message(), null);
+        }
+    }
+
+
+    @Nested
+    @DisplayName("retrieve tenant ID")
```
Comment (armiol) [inline server/src/test/java/io/spine/core/EventTest.java]: `expose tenant ID`

File: server/src/test/java/io/spine/core/EventTest.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    class GetTenantId {
+
+        @Test
+        @DisplayName("from event with command context")
```
Comment (armiol) [inline server/src/test/java/io/spine/core/EventTest.java]: `from enclosed command context`.

File: server/src/test/java/io/spine/core/EventTest.java [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+            EventContext context = contextWithoutOrigin().setCommandContext(commandContext)
+                                                         .build();
+            Event event = event(context);
+
+            assertThat(event.tenant())
+                    .isEqualTo(targetTenantId);
+        }
+
+        @Test
+        @DisplayName("from event with event context originated from command context")
```
Comment (armiol) [inline server/src/test/java/io/spine/core/EventTest.java]: `from enclosed context of origin event, which had command as origin`

**Pull request merged**